### PR TITLE
Add support for function.schedules

### PR DESCRIPTION
**Pull request by gregertw**
This pull request includes the following changes:

- Move requirements.txt file

- Working schedule loader

- Add tests for client.post and function schedules

### CODE REVIEW

**Review by doctrino — COMMENTED** [review]
A few code suggestions. Waiting on the deployment to test the deploy functionality.

**Comment by doctrino** [comment]
First time I ran `deploy` I got this failure
```
WARNING: Some resources were added due to dependencies.
Deploying 2 functions in 1 batch to CDF...
Deploying 2 functions.schedules in 1 batch to CDF...
ERROR: Failed to create resource(s).
Failed to bind session on schedule with id None. | code: 400 | X-Request-ID: 1fca53f0-f02d-9b9d-a5c5-1ce45ef81492
            Summary of Resources Deploy operation:
┏━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━┓
┃  Resource ┃ Created ┃ Deleted ┃ Changed ┃ Unchanged ┃ Total ┃
┡━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━┩
│ data_sets │       0 │       0 │       0 │         0 │     0 │
│ functions │       0 │       0 │       1 │         1 │     2 │
└───────────┴─────────┴─────────┴─────────┴───────────┴───────┘
ERROR:  Failure to load functions as expected.
```.
But rerunning it was successful. Trying to recreate it, but it takes time as deployment is takes a long time.

File: cognite_toolkit/cdf_tk/load/_resource_loaders.py [hunk 1]
```diff
@@ -505,6 +514,95 @@ def delete(self, ids: SequenceNotStr[str]) -> int:
...
+        schedules = FunctionSchedulesList([])
+        for func in functions:
+            ret = self.client.functions.schedules.list(function_id=func.id, limit=-1)
+            for schedule in ret:
+                schedule.function_external_id = func.external_id
+            schedules.extend(ret)
+        return schedules
+
+    def create(self, items: FunctionScheduleWriteList) -> FunctionSchedulesList:
+        items = self._resolve_functions_ext_id(items)
```
[the remark this answers is not in the document]
Comment (gregertw) [reply cognite_toolkit/cdf_tk/load/_resource_loaders.py:571]: Good point.

File: cognite_toolkit/cdf_tk/load/_resource_loaders.py [hunk 2]
```diff
@@ -505,6 +514,95 @@ def delete(self, ids: SequenceNotStr[str]) -> int:
...
+
+    @classmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
+        return FunctionsAcl([FunctionsAcl.Action.Read, FunctionsAcl.Action.Write], FunctionsAcl.Scope.All())
+
+    @classmethod
+    def get_id(cls, item: FunctionScheduleWrite | FunctionSchedule) -> str:
+        if item.function_external_id is None or item.cron_expression is None:
+            raise ValueError("FunctionSchedule must have functionExternalId and CronExpression set.")
+        return f"{item.function_external_id}:{item.cron_expression}"
```
[the remark this answers is not in the document]
Comment (gregertw) [reply cognite_toolkit/cdf_tk/load/_resource_loaders.py:538]: I'm not sure how that will help? I assume we don't want to extend the id concept and keep str? We could encapsulate the id better by having an ID class and a .as_string(), but I feel that is a bit overkill.

File: cognite_toolkit/cdf_tk/utils.py [hunk 3]
```diff
@@ -805,3 +806,34 @@ def calculate_directory_hash(directory: Path) -> str:
                     sha256_hash.update(chunk)
 
     return sha256_hash.hexdigest()
+
+
+def get_oneshot_session(
+    ToolGlobals: CDFToolConfig | None = None, client: CogniteClient | None = None
```
[the remark this answers is not in the document]
Comment (gregertw) [reply cognite_toolkit/cdf_tk/utils.py:812]: The Loaders are instantiated with CogniteClient only, so I don't have access to CDFToolConfig from inside the loaders. It's tempting to refactor loaders to be instantiated with CDFToolConfig instead...

[the remark this answers is not in the document]
Comment (gregertw) [reply cognite_toolkit/cdf_tk/utils.py]: I have then refactored this to not take either CDFToolConfig or CogniteClient, but just use CogniteClient. Cleans up and simplifies.

**Review by doctrino — APPROVED** [review]
I still have this warning:
```
Deploying 2 functions.schedules in 1 batch to CDF...
  WARNING: Failed to bind session because function is not ready.
 Batch 1/1 of functions.schedules: Created ['fn_example_repeater:0 8 * * *', 'fn_example_repeater:0 20 * * *']
```

**Comment by gregertw** [comment]
> I still have this warning:
> 
> ```
> Deploying 2 functions.schedules in 1 batch to CDF...
>   WARNING: Failed to bind session because function is not ready.
>  Batch 1/1 of functions.schedules: Created ['fn_example_repeater:0 8 * * *', 'fn_example_repeater:0 20 * * *']
> ```

Yes, if you do a clean and then a deploy, this is still present.

**Pull request merged**